src/main/java/com/cleanroommc/groovyscript/compat/mods/prodigytech/ExplosionFurnaceAdditives.java

```diff
@@ -0,0 +1,132 @@
+    @Override
+    public void onReload() {
+        removeScripted().forEach(EFAdditiveRecipe::unregister);
+        restoreFromBackup().forEach(EFAdditiveRecipe::register);
+    }
+
+    private void add(EFAdditiveRecipe recipe) {
+        recipe.register();
+        addScripted(recipe);
+    }
+
+    private void remove(EFAdditiveRecipe recipe) {
+        recipe.unregister();
+        addBackup(recipe);
+    }
+
+    @MethodDescription(example = @Example("item('minecraft:cobblestone'), 50"), type = MethodDescription.Type.ADDITION)
+    public void addExplosive(IIngredient explosive, int power) {
+        EFAdditiveRecipe recipe = new EFAdditiveRecipe(true, explosive, power);
+        add(recipe);
+    }
+
+    @MethodDescription(example = @Example("ore('gunpowder')"))
+    public boolean removeExplosive(IIngredient explosive) {
+        for (ItemStack it : explosive.getMatchingStacks()) {
+            ExplosionFurnaceManager.Explosive externalExplosive = ExplosionFurnaceManager.findExplosive(it);
+            if (externalExplosive == null) continue;
+            EFAdditiveRecipe recipe = new EFAdditiveRecipe(true, explosive, externalExplosive.getPower());
+            remove(recipe);
+            return true;
+        }
+        return false;
+    }
+
+    @MethodDescription(example = @Example(priority = 2000, commented = true))
+    public void removeAllExplosives() {
+        ExplosionFurnaceManager.EXPLOSIVES.getAllContent().forEach(r ->
+            addBackup(new EFAdditiveRecipe(true, new ItemsIngredient(r.getMatchingStacks()), r.getPower())));
+        ExplosionFurnaceManager.removeAllExplosives();
+    }
+
+    @MethodDescription(example = @Example("item('minecraft:stone'), 50"), type = MethodDescription.Type.ADDITION)
+    public void addDampener(IIngredient dampener, int power) {
+        EFAdditiveRecipe recipe = new EFAdditiveRecipe(false, dampener, power);
+        add(recipe);
+    }
+
+    @MethodDescription(example = @Example("ore('dustAsh')"))
+    public boolean removeDampener(IIngredient dampener) {
+        for (ItemStack it : dampener.getMatchingStacks()) {
+            ExplosionFurnaceManager.Dampener externalDampener = ExplosionFurnaceManager.findDampener(it);
+            if (externalDampener == null) continue;
+            EFAdditiveRecipe recipe = new EFAdditiveRecipe(false, dampener, externalDampener.getDampening());
+            remove(recipe);
+            return true;
+        }
+        return false;
+    }
+
+    @MethodDescription(example = @Example(priority = 2000, commented = true))
+    public void removeAllDampeners() {
+        ExplosionFurnaceManager.DAMPENERS.getAllContent().forEach(r ->
+            addBackup(new EFAdditiveRecipe(false, new ItemsIngredient(r.getMatchingStacks()), r.getDampening())));
+        ExplosionFurnaceManager.removeAllDampeners();
+    }
+
+    public static class EFAdditiveRecipe {

```

Comment:
I dont approve how this registry is done. Do not create a custom recipe class. Rather just use the existing explosive class and add a `AbstractReloadableStorage<Dampner>` field. See [this](https://github.com/CleanroomMC/GroovyScript/blob/ab64a4b26791f0848ae6100a001a72a1334c975f/src/main/java/com/cleanroommc/groovyscript/compat/mods/thermalexpansion/machine/Smelter.java#L25) for example

Reply:
wasn't aware of the existence of `AbstractReloadableStorage`

Reply:
Do I not inherit from `VirtualizedRegistry` then? Not sure if I need to do anything if my "registry" doesn't contain recipes in itself

Reply:
Ok so, I looked into the PT's code and it requires an accessor mixin. I feel like a less destructive option is better.

Reply:
Youd still have a `VirtualizedRegistry<Explossive>` with a field `AbstractReloadableStorage<Dampner>`. And there is nothing wrong with accessor mixins.

Reply:
made a version of this, unfortunately it is the only sensible version (the other version requires having a custom class too with all the consequences).